### PR TITLE
Move interface helper functions to `qml/math/interface_utils.py`

### PR DESCRIPTION
**Pull request by andrijapau**
**Context:**

Interface handling is currently scattered throughout the codebase. To standardize supported interfaces and streamline workflows, the first step is to centralize all related functions. `qml.math` is the natural choice as most of the interface management is currently being done through helpers found in `qml/math/utils.py`.

**Description of the Change:**

All interface handling functions independent of Pennylane structures (tapes, qnodes, etc) have been moved to `qml/math/interface_utils.py`.

Added new unit test file `test_resolve_interface.py` to improve testing of interface conversion logic in `execution.py`.

**Benefits:**

- Improved code organization.
- Easier to maintain and update interface handling logic.

**Possible Drawbacks:** None identified.

[sc-79357]

### CODE REVIEW

**Review by lillian542 — APPROVED** [review]
LGTM! 🎉

**Review by albi3ro — COMMENTED** [review]
For `jpc_interfaces`, I don't think that needs to be moved to `math`.  I've actually been thinking of hardcoding `if interface == "tensorflow-autograph"` in the source code instead.  Instead of managing which interfaces work with the newer framework, we can just hardcode the single interface that can't get ported.

File: pennylane/math/interface_utils.py
Comment (albi3ro) [inline pennylane/math/interface_utils.py:1]: Sorry to pivot on this, but we should try to keep anything that depends on PennyLane structures (like tapes) out of this module, even if it is an implicit dependency.

So for `_get_jax_interface_name` and `_resolve_interface`, those should probably go back in the workflow module.

Comment (andrijapau) [reply pennylane/math/interface_utils.py:1]: Done in latest commit 😄 , resolving.

File: tests/workflow/test_resolve_interface.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        pytest.skip("TensorFlow is not installed.")
+
+    # pylint: disable=not-context-manager
+    with tf.Graph().as_default():
+        tapes = [
+            QuantumScript([qml.RX(tf.constant(0.5), wires=0)], [qml.expval(qml.PauliZ(0))]),
+        ]
+        resolved_interface = _resolve_interface("tf", tapes)
+
+    assert resolved_interface == "tf-autograph"
```
Comment (andrijapau) [inline tests/workflow/test_resolve_interface.py:101]: @albi3ro, do you know of a better way to mimic a `tf-autograph` tape?

File: pennylane/math/__init__.py [hunk 2]
```diff
@@ -89,11 +89,17 @@
...
     convert_like,
-    get_deep_interface,
-    get_interface,
     in_backprop,
-    is_abstract,
     requires_grad,
+    is_abstract,
+)
+from .interface_utils import (
+    _get_canonical_interface_name,
```
Comment (albi3ro) [inline pennylane/math/__init__.py:97]: Potentially time to get rid of the `_` on the name and make the helper public.

Comment (andrijapau) [reply pennylane/math/__init__.py]: True. Resolving!

**Review by albi3ro — APPROVED** [review]
👍

File: tests/workflow/test_resolve_interface.py (hunk 1 above)
Comment (albi3ro) [reply tests/workflow/test_resolve_interface.py:101]: nope. autograph is just painful all around.

Comment (andrijapau) [reply tests/workflow/test_resolve_interface.py:101]: 😿 ... resolving.

**Pull request merged**